### DISCUSSION TITLE
How can I trigger a Github Action with the `afterChange` hook? - HTTP webhooks?

### DISCUSSION THREAD

**Discussion by gianpaj** [payloadcms/payload#1942] [Q&A] [answered]
Would it make sense to add a Setting in the Menu to easily add a URL to add a [Webhook](https://en.wikipedia.org/wiki/Webhook)?

Or should I hardcode the URL when I make the API call using the GitHub API to trigger a webhook event?
- https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#repository_dispatch
- https://blog.knoldus.com/manual-trigger-in-github-actions/

Context: I'm researching using this CMS to be able to trigger a static build of a Next.JS website made for SEO purposes.

**Comment by denolfe** [comment, marked as answer]
All configuration is typically done in the payload config and/or an .env file. This is where you would configure the URL and any other values.

For GitHub API, the way to interact programmatically would be to use [Octokit](https://github.com/octokit/octokit.js). So within your hook, you would instantiate Octokit, then use whatever methods they have available - likely something to do with a workflow run.

**Reply by gianpaj** [reply]
That's great! thanks so much:)
